src/hotspot/share/opto/type.hpp

```diff
@@ -26,6 +26,8 @@
 #define SHARE_OPTO_TYPE_HPP
 
 #include "opto/adlcVMDeps.hpp"
+#include "opto/compile.hpp"
```

Comment:
Please, don't include `compile.hpp` here - it could be cyclic dependencies if not now but later.
If you need something from it put it into `type.cpp`

Reply:
It is useless, we use `Compile::current()` inside `type.hpp`, which means that `compile.hpp` must be included before `type.hpp` anyway.

Reply:
`useless` -> `useful`
You are right. But how it works now? Usually C++ compilers complain about missing definitions.

Reply:
It seems that in all source files that include `type.hpp`, `compile.hpp` is also included either directly or indirectly before. So, there is no compilation error. It only arises when `rangeinference.cpp` only includes `type.hpp` and not many things else.